contrib/compare_ids_test.bzl

```diff
@@ -0,0 +1,80 @@
+# Copyright 2015 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Implementation of compare_ids_test
+def _compare_ids_test_impl(ctx):
+    tar_files = []
+    for tar in ctx.attr.tars:
+        tar_files += list(tar.files)
+
+    if (len(tar_files) == 0):
+        fail("No tar files provided for test.")
+
+    if (len(tar_files) == 1 and not ctx.attr.id):
+        fail("One tar provided. Need either second tar or an id to compare it to.")
+
+    tars_string = ""
+    for tar in tar_files:
+        tars_string += tar.short_path + " "
+
+    runfiles = ctx.runfiles(files = tar_files)
+
+    ctx.actions.expand_template(
+        template = ctx.file._executable_template,
+        output = ctx.outputs.executable,
+        substitutions = {"{id}": ctx.attr.id, "{tars}": tars_string},
+        is_executable = True,
+    )
+
+    return [DefaultInfo(runfiles = runfiles)]
+
+"""
+Test to compare ids of images in tarballs.
+Useful for testing reproducibility.
+
+Args:
+    tars: List of Labels which refer to the docker image tarballs (from docker save)
+    id: (optional) the id we want the images in the tarballs to have
+
+The test passes if all images in the tarballs have the given id.
+The test also passes if no id is provided and all tarballs have the same id.
+
+Each tarball must contain exactly one image.
+
+Examples of use:
+
+compare_ids_test(
+    name = "test1",
+    tars = ["image1.tar", "image2.tar", "image3.tar"],
+)
+
+compare_ids_test(
+    name = "test2",
+    tars = ["image.tar"],
+    id = "452a96d81c30a1e426bc250428263ac9ca3f47c9bf086f876d11cb39cf57aeec",
```

Comment:
one last comment,

replace this id to a dummy one like 

id = <my_image_sha256>